### PR TITLE
💄 Switch from Swagger UI to RapiDoc

### PR DESCRIPTION
**Pull request by afranke**
For a more modern interface to explore the API.

### CODE REVIEW

**Comment by afranke** [comment]
# Before

image

# After

image

# Notes

* RapiDoc supports custom API servers, but getting an entry like the one we had before requires adding metadata to the spec definition. I’ll take care of that in a separate PR to [the matrix-doc repository](https://github.com/matrix-org/matrix-doc/). I don’t consider this a blocker.
* I didn’t style RapiDoc with the website header/navbar **on purpose** because quite frankly it’s ugly in this context, gets in the way, and I doubt someone really needs to jump from the playground **directly** to the blog… The `[matrix]` logo in the top left corner is a link back to [the Develop section](https://matrix.org/docs/develop), where visitors can then find the navbar again.
* RapiDoc supports both Swagger 2.0 and OpenAPI 3.x.x, so we won’t have to change anything when we make that move eventually.

File: scripts/swagger-ui.patch [hunk 1]
```diff
@@ -1,80 +0,0 @@
---- api/client-server/index.html	2017-08-03 11:16:08.456850097 +0100
```
Comment (anoadragon453) [inline scripts/swagger-ui.patch:1]: Looks like [content/swagger.css](https://github.com/matrix-org/matrix.org/blob/master/content/swagger.css) can go as well.

File: scripts/update_docs.sh [hunk 2]
```diff
@@ -4,8 +4,8 @@
 # At present this includes:
```
Comment (anoadragon453) [inline scripts/update_docs.sh:4]: Should we change [scripts/update_docs_local.sh](https://github.com/matrix-org/matrix.org/blob/master/scripts/update_docs_local.sh) as well? Or does that file just seem redundant in general (and we should remove it)?

File: rapidoc/index.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<!doctype html> <!-- Important: must specify -->
```
Comment (anoadragon453) [inline rapidoc/index.html:1]: For instances like:

https://github.com/matrix-org/matrix.org/blob/a53ecfc9dad7afc939aab96ff5a903bb8c64f51e/unstyled_docs/spec/index.html#L1111

Would we still call RapiDoc a "Swagger Viewer", or should we switch entirely to the terms OpenAPI/RapiDoc?

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
Some minor clarifying points, but otherwise looks good to me.

Design may want to modify the colour scheme at some point, but from your screenshot I think it looks pretty slick for a first pass (though I have no idea about accessibility for instance).

File: rapidoc/index.html (hunk 3 above)
Comment (afranke) [reply rapidoc/index.html:1]: I’d say OpenAPI indeed. Lemme fix it real quick.

File: scripts/update_docs.sh (hunk 2 above)
Comment (afranke) [reply scripts/update_docs.sh:4]: I have actually never used it, and I suspect other things in the scripts are not in sync either. Whether we remove it or get it more in line with the non local one, we can do this in another PR?

File: scripts/swagger-ui.patch (hunk 1 above)
Comment (afranke) [reply scripts/swagger-ui.patch:1]: Gone.

File: scripts/update_docs.sh (hunk 2 above)
Comment (afranke) [reply scripts/update_docs.sh:4]: I had a quick look, it’s indeed out of sync and is basically a(n old) copy with some commented out steps. I reckon a better solution would be to remove the script altogether and make some steps optional in the main one.

Out of scope here.

Comment (anoadragon453) [reply scripts/update_docs.sh:4]: Yep, that was my hunch and thanks for checking. Yes, we can clean it up separately.

**Review by anoadragon453 — APPROVED** [review]

**Pull request merged**